frontend/src/stores/courses.js

```diff
@@ -157,6 +157,7 @@ class courseStore {
                 });
                 this.rootStore.userStore.setBalance(course.price);
                 reservation.bookingStatus = 1;
+                await this.fetchCourses();
```

Comment:
can you simple reduce the number of courses using the store. This would add more delay to the reservation flow. what do you think?. Also we do have another fetchCourses running every 5 seconds and that should update this later. 

Say there are 5 seats left.
User books 1. You can just substract the available seats to `availableseats - 1 ` and update UI.
And we have the check running at 5 second intervals and you can grab the available seats from that request and set to the store again. So that if some one else had made the reservation then the count would be 3.

Reply:
Yes, this is not the ideal way. I'll be taking care of this once all the other tickets are closed